src/Features/CSharp/Portable/Diagnostics/Analyzers/FileBasedPrograms/FileLevelDirectiveDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.CodeQuality;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.DotNet.FileBasedPrograms;
+
+namespace Microsoft.CodeAnalysis.FileBasedPrograms;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+internal sealed class FileLevelDirectiveDiagnosticAnalyzer()
+    : AbstractCodeQualityDiagnosticAnalyzer(
+        descriptors: [Rule],
+        generatedCodeAnalysisFlags: GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics)
+{
+    public const string DiagnosticId = "FileBasedPrograms";
+
+    private static readonly DiagnosticDescriptor Rule = CreateDescriptor(
```

Comment:
Should private static fields follow this convention?
```suggestion
    private static readonly DiagnosticDescriptor s_rule = CreateDescriptor(
```